### PR TITLE
libcontainer/apparmor: use sync.Once for AppArmor detection

### PR DESCRIPTION
**Pull request by thaJeztah**
I recalled I added a similar change in containerd, but didn't upstream it here. Also removing `bytes.HasPrefix()` in favor of just checking the first character

### CODE REVIEW

**Review by AkihiroSuda — DISMISSED** [review]

File: libcontainer/apparmor/apparmor_linux.go [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 	"fmt"
 	"io/ioutil"
 	"os"
+	"sync"
 
 	"github.com/opencontainers/runc/libcontainer/utils"
 )
 
+var (
+	appArmorSupported bool
```
Comment (kolyshkin) [inline libcontainer/apparmor/apparmor_linux.go:14]: nit: `s/Supported/Enabled/`

**Comment by kolyshkin** [comment]
A nit, LGTM otherwise

File: libcontainer/apparmor/apparmor_linux.go (hunk 1 above)
Comment (thaJeztah) [reply libcontainer/apparmor/apparmor_linux.go]: Dang! I looked at it and wanted to update, but forgot: done 👍

**Comment by kolyshkin** [comment]
CI failure is a flake (https://github.com/opencontainers/runc/issues/2760#issuecomment-800444105), CI restarted.

**Review by kolyshkin — APPROVED** [review]
LGTM

**Comment by kolyshkin** [comment]
CI failure
```
not ok 26 global --debug to --log
# (in test file tests/integration/debug.bats, line 40)
#   `[[ "${output}" == *"child process in init()"* ]]' failed
```
is a known flake (#2756, should be fixed by #2835).

Restarted CI.

**Comment by kolyshkin** [comment]
This PR appears to be a champion in the amount of unrelated CI failures.

https://github.com/opencontainers/runc/issues/2861

CI restarted

**Comment by thaJeztah** [comment]
> This PR appears to be a champion in the amount of unrelated CI failures.
> 
> #2861
> 
> CI restarted

I opened this PR to reveal flaky tests 🤣😂

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**